pkg/types/ccipocr3/chain_accessor.go

```diff
@@ -0,0 +1,389 @@
+	// Contract: OffRamp
+	// Confidence: Unconfirmed
+	GetLatestPriceSeqNr(ctx context.Context) (uint64, error)
+
+	// GetOffRampConfigDigest returns the offramp config digest for the provided plugin type.
+	//
+	// TODO: this should go away, it's a call to the ConfigPoller.
+	GetOffRampConfigDigest(ctx context.Context, pluginType uint8) ([32]byte, error)
+
+	// GetOffRampSourceChainsConfig returns the sourceChains config for all the provided source chains.
+	// If a config was not found it will be missing from the returned map.
+	//
+	// TODO: this should go away, it's a call to the ConfigPoller.
+	GetOffRampSourceChainsConfig(
+		ctx context.Context,
+		sourceChains []ChainSelector,
+	) (map[ChainSelector]SourceChainConfig, error)
+}
+
+type SourceAccessor interface {
+	// MsgsBetweenSeqNums returns all messages being sent to the provided dest
+	// chain, between the provided sequence numbers. Messages are sorted ascending
+	// based on their timestamp.
+	//
+	// Access Type: Event(CCIPMessageSent)
+	// Contract: OnRamp
+	// Confidence: Finalized
+	MsgsBetweenSeqNums(
+		ctx context.Context,
+		dest ChainSelector,
+		seqNumRange SeqNumRange,
+	) ([]Message, error)
+
+	// LatestMsgSeqNum returns the sequence number associated with the most
+	// recent message being sent to a given destination.
+	//
+	// Access Type: Event(CCIPMessageSent)
+	// Contract: OnRamp
+	// Confidence: Finalized
+	//
+	// TODO: Rename function in CAL to LatestMessageTo(dest) and return the entire message.
+	//       See Design Doc (LatestMsgSeqNum) for notes.
+	LatestMsgSeqNum(ctx context.Context, dest ChainSelector) (SeqNum, error)
+
+	// GetExpectedNextSequenceNumber returns the expected next sequence number
+	// messages being sent to the provided destination.
+	//
+	// Access Type: Method(GetExpectedNextSequenceNumber)
+	// Contract: OnRamp
+	// Confidence: Unconfirmed
+	GetExpectedNextSequenceNumber(
+		ctx context.Context,
+		dest ChainSelector,
+	) (SeqNum, error)
+
+	// GetTokenPriceUSD looks up a token price in USD. The address value should
+	// be retrieved from a configuration cache (i.e. ConfigPoller).
+	//
+	// Access Type: Method(GetTokenPrice)
+	// Contract: FeeQuoter
+	// Confidence: Unconfirmed
+	//
+	// Notes: This function is new and serves as a general price interface for
+	//        both LinkPriceUSD and GetWrappedNativeTokenPriceUSD.
+	//        See Design Doc (Combined Token Price Helper) for notes.
+	GetTokenPriceUSD(
+		ctx context.Context,
+		address UnknownAddress,
+	) (BigInt, error)
+
+	// GetFeeQuoterDestChainConfig returns the fee quoter destination chain config.
+	//
+	// Access Type: Method(GetDestChainConfig)
+	// Contract: FeeQuoter
+	// Confidence: Unconfirmed
+	//
+	// Notes: This is a new general purpose function needed to implement
+	//        GetMedianDataAvailabilityGasConfig.
+	GetFeeQuoterDestChainConfig(
+		ctx context.Context,
+		dest ChainSelector,
+	) (FeeQuoterDestChainConfig, error)
+}
+
+type RMNAccessor interface {
```

Comment:
These are only on the destination chain, so this could be a subcomponent for the DestinationAccessor.